### PR TITLE
Fix Flink integration test

### PR DESCRIPTION
**Pull request by medb**
To fix failure in HA mode this PR starts Flink Master with YARN session only on first master node.

Failure:
```
======================================================================
FAIL: test_flink.mode: HA.version: 1_2.random_prefix: blo9 (flink.test_flink.FlinkTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/google/home/idv/.local/lib/python3.6/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/usr/local/google/home/idv/dev/src/dataproc-initialization-actions/flink/test_flink.py", line 52, in test_flink
    self.createCluster(configuration, self.INIT_ACTIONS, dataproc_version, metadata=metadata)
  File "/usr/local/google/home/idv/dev/src/dataproc-initialization-actions/integration_tests/dataproc_test_case.py", line 109, in createCluster
    "Failed to create Cluster {}. Error: {}".format(self.name, stderr))
AssertionError: -9 != 0 : Failed to create Cluster test-flink-ha-1-2-20190608-213723-6u5y. Error: Waiting on operation [projects/google.com:hadoop-cloud-dev/regions/global/operations/a64538ba-2353-3a38-bd06-8400c9197251].
Waiting for cluster creation operation...
WARNING: For PD-Standard, we strongly recommend provisioning 1TB or larger to ensure consistently high I/O performance. See https://cloud.google.com/compute/docs/disks/performance for information on disk I/O performance.
```

### CODE REVIEW

File: flink/flink.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@
 # Cloud Dataproc Image Version information:
 # https://cloud.google.com/dataproc/concepts/dataproc-versions
 
-set -euxo pipefail
+set -Eeuxo pipefail
```
Comment (functicons) [inline flink/flink.sh]: Why this flag?

File: flink/flink.sh [hunk 2]
```diff
@@ -92,7 +92,8 @@ function install_flink_snapshot() {
   tar -xzvf "${flink_local}"
   rm "${flink_local}"
   # only the first match of the flink toplevel pattern is used
-  local files=( ${flink_toplevel_pattern} )
+  local files
+  mapfile -t files < <(compgen -G "${flink_toplevel_pattern}")
```
Comment (functicons) [inline flink/flink.sh]: This is too hard to understand.

File: flink/flink.sh [hunk 3]
```diff
@@ -53,11 +53,9 @@ readonly START_FLINK_YARN_SESSION_DEFAULT=true
 # Set this to install flink from a snapshot URL instead of apt
 readonly FLINK_SNAPSHOT_URL_METADATA_KEY='flink-snapshot-url'
 
-
-
 function err() {
-  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
-  return 1
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $*" >&2
+  exit 1
```
[the remark this answers is not in the document]
Comment (medb) [reply flink/flink.sh]: Done

File: flink/flink.sh [hunk 4]
```diff
@@ -76,13 +74,15 @@ function update_apt_get() {
 }
 
 function install_apt_get() {
-  pkgs="$@"
+  pkgs="$*"
```
[the remark this answers is not in the document]
Comment (medb) [reply flink/flink.sh]: Done.

File: flink/flink.sh (hunk 1 above)
Comment (medb) [reply flink/flink.sh]: This is `errtrace` flag, removed because we don't use `trap` anyway

File: flink/flink.sh (hunk 2 above)
Comment (medb) [reply flink/flink.sh]: This is idiom recommended by https://github.com/koalaman/shellcheck

Comment (functicons) [inline flink/flink.sh]: Break it down into 2 steps:

compgen -G "${flink_toplevel_pattern}" >/tmpfile
mapfile -t files </tmpfile

Comment (medb) [reply flink/flink.sh]: Done.

**Pull request merged**